### PR TITLE
48098_Update_Profile with modification of Sign out, better session ma…

### PR DESCRIPTION
**Pull request by AVI5HEK**
…nagement and update

1. remember_token added
2. remember_me added back

### CODE REVIEW

File: app/src/main/java/com/framgia/elsytem/LoginActivity.java [hunk 1]
```diff
@@ -78,7 +79,7 @@ public void onClick(View v) {
                     if (checkBoxRememberMe.isChecked()) mRememberMe = 1;
                     else mRememberMe = 0;
                     if (!isConnected())
-                        alert.showAlertDialog(getApplicationContext(), getString(R.string
+                        mShowDialog(LoginActivity.this, getString(R.string
                                         .connection_error_title_activity_login),
                                 getString(R.string.connection_error_message_activity_login), false);
                     if (!mEmail.equals("") && !mPassword.equals("")) {
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/LoginActivity.java]: you can you this util class: TextUtils.isEmpty(String)

File: app/src/main/java/com/framgia/elsytem/LoginActivity.java [hunk 2]
```diff
@@ -94,12 +95,43 @@ public void onClick(View v) {
...
+        if (status != null)
+            // Setting alert dialog icon
+            alertDialog.setIcon((status) ? R.drawable.ico_success : R.drawable.ico_fail);
+        alertDialog.setButton(DialogInterface.BUTTON_NEUTRAL, "Cancel", new
+                DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                    }
+                });
+        alertDialog.setButton(DialogInterface.BUTTON_POSITIVE, "Turn on Wi-Fi", new DialogInterface.OnClickListener() {
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/LoginActivity.java]: do not hard code

File: app/src/main/java/com/framgia/elsytem/LoginActivity.java [hunk 3]
```diff
@@ -94,12 +95,43 @@ public void onClick(View v) {
...
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                startActivity(new Intent(Settings.ACTION_WIFI_SETTINGS));
+            }
+        });
+        alertDialog.setButton(DialogInterface.BUTTON_NEGATIVE, "Turn on cellular data", new DialogInterface.OnClickListener() {
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/LoginActivity.java]: hardcode

File: app/src/main/java/com/framgia/elsytem/LoginActivity.java [hunk 4]
```diff
@@ -161,14 +193,15 @@ protected void onPostExecute(String result) {
...
+                    String rememberToken = "";
                     //creating session
                     if (!name.isEmpty() && !email.isEmpty())
-                        session.createLoginSession(name, email, avatar, rememberToken);
+                        session.createLoginSession(name, email, avatar, rememberToken, mRememberMe);
                     Log.e(TAG + " Name: ", name);
                     Log.e(TAG + " Email: ", email);
                     Log.e(TAG + " Avatar: ", avatar);
-                    Log.e(TAG + " Remembr: ", rememberToken);
+                    Log.e(TAG + " Token: ", rememberToken);
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/LoginActivity.java]: too many Log.e if you want to log you can use Log.e(TAG + " Name: ", name + " / Email: " + email .....);

File: app/src/main/java/com/framgia/elsytem/mypackage/AlertDialogManager.java [hunk 5]
```diff
@@ -22,23 +23,18 @@
...
-
         // Setting Dialog Message
         alertDialog.setMessage(message);
-
         if (status != null)
             // Setting alert dialog icon
             alertDialog.setIcon((status) ? R.drawable.ico_success : R.drawable.ico_fail);
-
         // Setting OK Button
         alertDialog.setButton("OK", new DialogInterface.OnClickListener() {
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/mypackage/AlertDialogManager.java]: hard code

File: app/src/main/java/com/framgia/elsytem/LoginActivity.java (hunk 1 above)
Comment (AVI5HEK) [inline app/src/main/java/com/framgia/elsytem/LoginActivity.java]: Can I use (!mEmail.isEmpty() && !mPassword.isEmpty()) instead?

**Pull request merged**